startup.py

```diff
@@ -0,0 +1,373 @@
+            "/Applications/Nuke{version}/{variant}{version_back}{suffix}.app",
+        ],
+        "win32": [
+            # C:\Program Files\Nuke10.0v5\Nuke10.0.exe
+            "C:/Program Files/Nuke{version}/Nuke{major_minor_version}.exe",
+        ],
+        "linux2": [
+            # /usr/local/Nuke10.0v5/Nuke10.0
+            "/usr/local/Nuke{version}/Nuke{major_minor_version}",
+            # /home/<username>/Nuke10.0v5/Nuke10.0
+            os.path.expanduser("~/Nuke{version}/Nuke{major_minor_version}")
+        ]
+    }
+
+    def _get_icon_from_product(self, product):
+        """
+        Returns the icon based on the product.
+
+        :param str product: Product name.
+
+        :returns: Path to the product's icon.
+        """
+        if "studio" in product.lower():
+            return os.path.join(
+                self.disk_location,
+                "icon_nukestudio_256.png"
+            )
+        elif "hiero" in product.lower():
+            return os.path.join(
+                self.disk_location,
+                "icon_hiero_256.png"
+            )
+        elif "nukex" in product.lower():
+            return os.path.join(
+                self.disk_location,
+                "icon_x_256.png"
+            )
+        else:
+            return os.path.join(
+                self.disk_location,
+                "icon_256.png"
+            )
+
+    def _find_executables(self, match_templates):
+        """
+        Finds all the executables available for this platform.
+
+        :param list match_templates: List of Glob patterns that can match Nuke executables.
+
+        :returns: Dictionary of results where the glob pattern is the key and the value
+            is the files found with said pattern.
+        """
+
+        # build up a dictionary where the key is the match template and the
+        # value is a list of matching executables. we'll need to keep the
+        # association between template and matches for later when we extract
+        # the components (version and product)
+        executable_matches = {}
+        for match_template in match_templates:
+
+            # build the glob pattern by formatting the template for globbing
+            glob_pattern = _format(match_template, dict((key, "*") for key in self.COMPONENT_REGEX_LOOKUP))
+            self.logger.debug(
+                "Globbing for executable matching: %s ..." % (glob_pattern,)
+            )
+            matching_paths = glob.glob(glob_pattern)
+            if matching_paths:
+                # found matches, remember this association (template: matches)
+                executable_matches[match_template] = [x.replace("\\", "/") for x in matching_paths]
+                self.logger.debug(
+                    "Found %s matches: %s" % (
+                        len(matching_paths),
+                        matching_paths
+                    )
+                )
+
+        return executable_matches
+
+    def _find_products(self):
```

Comment:
suggest `_find_software_versions`. at least until we have a better name for `SoftwareVersion` object.